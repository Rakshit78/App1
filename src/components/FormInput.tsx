import { Button, Container, TextField, Typography } from '@mui/material';
import { Box } from '@mui/system';
import { useState } from 'react';

type Props = {
  navigate: any;
};
const FormInput: React.FC<Props> = ({ ...props }) => {
  const [name, setname] = useState('');

  const handleclick = () => {
    props.navigate(`/details?name=${name}`);
  };
  return (
    <div
      style={{
        width: '100vw',
        height: '100vh',
        display: 'grid',
        placeItems: 'center',
      }}
    >
      <Container
        maxWidth='sm'
        style={{ background: '#e5e5e5', padding: '14px', borderRadius: '14px' }}
      >
        <Box ml={1}>
          <Typography variant='h5' style={{ opacity: '0.5' }}>
            Search Country
          </Typography>
        </Box>
        <Box mt={2} p={1}>
          <TextField
            fullWidth
            placeholder='Enter Coutry Name'
            role='takeinput'
            label='Input'
            value={name}
            onChange={(e: any) => {
              setname(e.target.value);
            }}
          />
        </Box>
        <Box mt={2} p={1}>
          <Button
            variant='contained'
            role='btn'
            fullWidth
            onClick={handleclick}
          >
            Get
          </Button>
        </Box>
      </Container>
    </div>
  );
};

export default FormInput;
