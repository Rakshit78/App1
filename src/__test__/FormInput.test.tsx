import React from 'react';
//import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import FormInput from '../components/FormInput';

describe('Form input test', () => {
  test('render formInput without crash', () => {
    render(<FormInput navigate={undefined} />);
  });

  test('input changes', () => {
    render(<FormInput navigate={undefined} />);
    expect(screen.getByLabelText('Input')).toBeInTheDocument();
  });

  test('input changes', () => {
    const component = render(<FormInput navigate={undefined} />);

    expect(component).toMatchSnapshot();
  });
});
