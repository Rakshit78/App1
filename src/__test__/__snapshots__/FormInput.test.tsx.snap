// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form input test input changes 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="width: 100vw; height: 100vh; display: grid; place-items: center;"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthSm css-cuefkz-MuiContainer-root"
          style="background: rgb(229, 229, 229); padding: 14px; border-radius: 14px;"
        >
          <div
            class="MuiBox-root css-1isemmb"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
              style="opacity: 0.5;"
            >
              Search Country
            </h5>
          </div>
          <div
            class="MuiBox-root css-10wfmn8"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              role="takeinput"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":r2:"
                id=":r2:-label"
              >
                Input
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r2:"
                  placeholder="Enter Coutry Name"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-1ftyaf0"
                  >
                    <span>
                      Input
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root css-10wfmn8"
          >
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root  css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
              role="btn"
              tabindex="0"
              type="button"
            >
              Get
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="width: 100vw; height: 100vh; display: grid; place-items: center;"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthSm css-cuefkz-MuiContainer-root"
        style="background: rgb(229, 229, 229); padding: 14px; border-radius: 14px;"
      >
        <div
          class="MuiBox-root css-1isemmb"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            style="opacity: 0.5;"
          >
            Search Country
          </h5>
        </div>
        <div
          class="MuiBox-root css-10wfmn8"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            role="takeinput"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for=":r2:"
              id=":r2:-label"
            >
              Input
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r2:"
                placeholder="Enter Coutry Name"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1ftyaf0"
                >
                  <span>
                    Input
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-10wfmn8"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root  css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
            role="btn"
            tabindex="0"
            type="button"
          >
            Get
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
